src/server/game/DataStores/DB2Stores.cpp

```diff
@@ -2115,6 +2133,33 @@ int32 const* DB2Manager::GetBroadcastTextDuration(uint32 broadcastTextId, Locale
     return Trinity::Containers::MapGetValuePtr(_broadcastTextDurations, { broadcastTextId, WowLocaleToCascLocaleBit[locale] });
 }
 
+QuestLineXQuestEntry const* DB2Manager::GetQuestLineXQuestForQuest(uint32 questId) const
+{
+    for (auto const& [questLineId, quests] : _questsByQuestXLine)
+    {
+        for (QuestLineXQuestEntry const* questLineXQuest : quests)
+            if (questLineXQuest->QuestID == questId)
```

Comment:
No need to check that.

In addition, you only return the first result and you have cases like quest 48111 that has two questlines (and only one is linked to a campaign). So depending on which one is returned, the quest could be considered as part of a campaign or not (you need to check if the player can have that campaign active)